data/ui/PreferencesPithosDialog.ui

```diff
@@ -168,14 +168,29 @@
                         <property name="column_spacing">20</property>
                         <property name="row_homogeneous">1</property>
                         <child>
+                          <object class="GtkCheckButton" id="use_downloadbuffering_checkbutton">
```

Comment:
I don't like internal implementation details being options.

Reply:
I know. The problem is that although the downloadbuffer element seems to work great and really is an improvement over memory buffering. There's no way to detect programmatically if it screws up as it will just cause Pithos to lockup. This gives users of older versions of Gstreamer an out in that situation.

Reply:
I also really don't like us storing this on disk, it opens up an obvious piracy problem.

> This gives users of older versions of Gstreamer an out in that situation.

If new gstreamer fixes problems we can just make it a hard dependency, especially with flatpak that isn't a problem.

Reply:
It's only cached for as long as needed. The file is automatically deleted when the song is over/skipped.

Reply:
Really though. I'm not married to the idea completely. What you ever you'd like done is OK with me. At this point we have 3 choices:

1. Leave it like it is in the PR.
2. Scrap the idea and just stick with in memory buffering.
3. Do like you suggested and set a (yet to be decided) version of Gstreamer as a hard dependency and get rid of the user option.

Reply:
I would prefer 2/3 and downloading can be discussed separately from this which should just be a refactoring mostly.

Reply:
> I would prefer 2/3 and downloading can be discussed separately from this which should just be a refactoring mostly.

OK, put it on hold for now then it is. Easy enough.

Reply:
I'll roll back to moving the player logic and tweak from there as far as the custom pipeline with a downloadbuffer for now so it'll be 2 clean commits.